pkgs/development/python-modules/withings-api/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchPypi
+, poetry-core
+, arrow
+, requests-oauthlib
+, typing-extensions
+, pydantic
+, unittestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "withings-api";
+  version = "2.4.0";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchPypi {
+    pname = "withings_api";
+    inherit version;
+    sha256 = "sha256-vQ6MKeD4g4QTkXx638FW53mTkx78af7NQXF00kxgM10=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace 'requests-oauth = ">=0.4.1"' '''
+  '';
+
+  propagatedBuildInputs = [
+    poetry-core
+    arrow
+    requests-oauthlib
+    typing-extensions
+    pydantic
+  ];
+
+  checkInputs = [
+    unittestCheckHook
```

Comment:
```
Executing unittestCheckPhase

----------------------------------------------------------------------
Ran 0 tests in 0.000s
```

Make sure we also fetch the tests. If they're not in the pypi sdist, then we need to use `fetchFromGitHub`.

Reply:
Moved to fetchFromGitHub to have testing.